core-test/src/test/scala/org/bitcoins/core/psbt/PSBTTest.scala

```diff
@@ -476,4 +491,160 @@ class PSBTTest extends BitcoinSAsyncTest {
       assert(signedPsbt1 == expectedPsbt1)
     }
   }
+
+  def dummyTx(
+      scriptSig: ScriptSignature = EmptyScriptSignature,
+      spk: ScriptPubKey = EmptyScriptPubKey): Transaction = {
+    BaseTransaction(
+      version = Int32.zero,
+      inputs = Vector(
+        TransactionInput(outPoint =
+                           TransactionOutPoint(txId = DoubleSha256Digest.empty,
+                                               vout = UInt32.zero),
+                         scriptSignature = scriptSig,
+                         sequenceNumber = UInt32.zero)),
+      outputs = Vector(TransactionOutput(CurrencyUnits.oneBTC, spk)),
+      lockTime = UInt32.zero
+    )
+  }
+
+  def dummyPSBT(
+      scriptSig: ScriptSignature = EmptyScriptSignature,
+      spk: ScriptPubKey = EmptyScriptPubKey): PSBT = {
+    PSBT.fromUnsignedTx(dummyTx(scriptSig, spk))
+  }
+
+  it must "successfully change a NonWitnessUTXO to a WitnessUTXO when compressing" in {
+    val wspk = P2WSHWitnessSPKV0(EmptyScriptPubKey)
+
+    val output =
+      TransactionOutput(value = CurrencyUnits.oneBTC, scriptPubKey = wspk)
+
+    val tx = dummyTx(spk = wspk)
+    val inputMap =
+      InputPSBTMap(Vector(InputPSBTRecord.NonWitnessOrUnknownUTXO(tx)))
+
+    val expectedInputMap =
+      InputPSBTMap(Vector(InputPSBTRecord.WitnessUTXO(output)))
+
+    val input = TransactionInput(
+      outPoint = TransactionOutPoint(txId = tx.txId, vout = UInt32.zero),
+      scriptSignature = EmptyScriptSignature,
+      sequenceNumber = UInt32.zero)
+
+    val compressedInputMap = inputMap.compressMap(input)
+
+    assert(compressedInputMap == expectedInputMap)
+  }
+
+  it must "do nothing when compressing a finalized InputPSBTMap" in {
+    val finalizedInputMap = InputPSBTMap(
+      Vector(InputPSBTRecord.FinalizedScriptSig(EmptyScriptSignature)))
+
+    val dummyInput = dummyTx().inputs.head
+
+    val compressedInputMap = finalizedInputMap.compressMap(dummyInput)
```

Comment:
Should just be able to use `EmptyTransactionInput` instead of all this `dummyTx` stuff